### ISSUE TITLE
Insteon Remote Control On/Off Switch displays wrong state randomly

### ISSUE THREAD

**Issue by efthymakis** [kuestess/homebridge-platform-insteonlocal#140]
The Insteon Remote Control On/Off Switch (2477S) randomly shows a status of 66% or 33%.  The only valid states are 0% or 100% as the switch is not dimmable -- it's either on or off.

The device is displayed correctly (other than its status).  Turning the device on or off through the Home app otherwise works fine.  I have the switch attached to a fan, so I've defined it as such.  I've tried changing the config to a switch, both dimmable and not, witch no change -- the stays still randomly reflects a percentage.

When changing the status from 66% to off, I see the following in the log:

4/26/2021, 10:15:36 PM [InsteonLocal] Got updated status for Fan
4/26/2021, 10:15:36 PM [InsteonLocal] Getting state for Fan

Device Config:

"name": "Fan",
            "deviceID": "4352AB",
            "deviceType": "fan",
            "dimmable": "no"

**Comment by kuestess** [comment]
@efthymakis Definitely need to configure as a switch - status messages returned from the hub have different meanings for switch vs fan.  Can you run homebridge in debug mode `homebridge -d` and post the console output when you get the random % status (not on or off).

**Comment by efthymakis** [comment]
Not sure I captured what you're looking for, but the issue might be resolved either way -- I'll explain at the end.  I am running Hoobs 3.3.5, not homebridge directly.  As such, I stopped the hoops service and then restarted it in debug mode.  I don't think it actually puts homebridge in debug since the log does not look much different :( .

I left the configuration as-is, with the device type defined as a "fan". I captured the log after I restarted the service and noticed the state of the 'fan' device as 66% in the Home app UI.  I then turned the device off, on, then off again.  I then waited for the service to refresh the status in the next polling cycle and it immediately reflected as 66% in the Home app UI.

5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Bathroom Light
5/3/2021, 11:03:36 PM [InsteonLocal] Getting state for Fan
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Hallway
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Foyer
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Front Landscape
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Holiday
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Porch
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Garage Light
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Kitchen Light
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Kitchen Table
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Under Cabinet
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Living Room Light
5/3/2021, 11:03:36 PM [InsteonLocal] Getting power state for Christmas Tree
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Dining Room Light
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Entrance
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Sitting Room Light
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Hall
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Hallway Secondary
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Entrance Secondary
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Kitchen Secondary
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Side Entrance Secondary
5/3/2021, 11:03:36 PM [InsteonLocal] Getting status for Hall Secondary
5/3/2021, 11:03:36 PM [InsteonLocal] Insteon event listener started...
5/3/2021, 11:03:36 PM [InsteonLocal] Started Insteon Express server...
5/3/2021, 11:03:37 PM [Ring] Found the following locations:
5/3/2021, 11:03:37 PM [Ring]   locationId: dr5xbg-2tggj-0 - LOCATION
5/3/2021, 11:03:37 PM [Ring] Creating location socket.io connection - LOCATION
5/3/2021, 11:03:37 PM [Ring] Ring connected to socket.io server
5/3/2021, 11:03:37 PM [Ring] Configuring 1 cameras and 4 devices for location "LOCATION" - locationId: dr5xbg-2tggj-0
5/3/2021, 11:03:37 PM [Ring] Hidden accessory 85aa1b77-3f0e-4f3e-b54a-a76853693c15 group.light-group.beams Front
5/3/2021, 11:03:41 PM [InsteonLocal] Got updated status for Basement Front
5/3/2021, 11:03:48 PM [InsteonLocal] Got updated status for Entrance
5/3/2021, 11:03:50 PM [InsteonLocal] Got updated status for Front Landscape
5/3/2021, 11:03:53 PM [InsteonLocal] Error getting power state of Dining Room Light
5/3/2021, 11:04:06 PM [InsteonLocal] Got updated status for Kitchen Secondary
5/3/2021, 11:04:18 PM [HOOBS CAD8@Fan@Fan@On] SET handler returned write response value, though the characteristic doesn't support write response
5/3/2021, 11:04:19 PM [InsteonLocal] Setting speed of Fan to off (false%)
5/3/2021, 11:04:21 PM [HOOBS CAD8@Fan@Fan@On] SET handler returned write response value, though the characteristic doesn't support write response
5/3/2021, 11:04:21 PM [HOOBS CAD8@Fan@Fan@Rotation Speed] SET handler returned write response value, though the characteristic doesn't support write response
5/3/2021, 11:04:22 PM [InsteonLocal] Setting speed of Fan to high (100%)
5/3/2021, 11:04:26 PM [HOOBS CAD8@Fan@Fan@On] SET handler returned write response value, though the characteristic doesn't support write response
5/3/2021, 11:04:26 PM [InsteonLocal] Setting speed of Fan to off (false%)
5/3/2021, 11:06:30 PM [InsteonLocal] Got updated status for Patio
5/3/2021, 11:06:35 PM [InsteonLocal] Got updated status for Deck
5/3/2021, 11:06:35 PM [InsteonLocal] Got updated status for Sitting Room Light
5/3/2021, 11:06:35 PM [InsteonLocal] Got updated status for Foyer
5/3/2021, 11:06:37 PM [InsteonLocal] Got updated status for Under Cabinet
5/3/2021, 11:06:38 PM [InsteonLocal] Got off event for Kitchen Secondary
5/3/2021, 11:06:38 PM [InsteonLocal] Got off event for Kitchen Secondary
5/3/2021, 11:06:43 PM [InsteonLocal] Got off event for Kitchen Secondary
5/3/2021, 11:06:43 PM [InsteonLocal] Got off event for Kitchen Secondary
5/3/2021, 11:06:44 PM [InsteonLocal] Got updated status for Living Room Light
5/3/2021, 11:06:46 PM [InsteonLocal] Got updated status for Garage Light
5/3/2021, 11:06:48 PM [InsteonLocal] Got updated status for Hall
5/3/2021, 11:06:50 PM [InsteonLocal] Got updated status for Side Entrance Secondary
5/3/2021, 11:06:52 PM [InsteonLocal] Got updated status for Kitchen Secondary
5/3/2021, 11:06:57 PM [InsteonLocal] Got updated status for Kitchen Secondary
5/3/2021, 11:06:59 PM [InsteonLocal] Got updated status for Christmas Tree
5/3/2021, 11:06:59 PM [InsteonLocal] Getting power state for Christmas Tree
5/3/2021, 11:07:01 PM [InsteonLocal] Got updated status for Bathroom Light
5/3/2021, 11:07:03 PM [InsteonLocal] Got updated status for Entrance
5/3/2021, 11:07:03 PM [InsteonLocal] Got updated status for Holiday
5/3/2021, 11:07:10 PM [InsteonLocal] Got updated status for Kitchen Table
5/3/2021, 11:07:14 PM [InsteonLocal] Got updated status for Side Entrance
5/3/2021, 11:07:16 PM [InsteonLocal] Got updated status for Porch
5/3/2021, 11:07:17 PM [InsteonLocal] Got updated status for Entrance Secondary
5/3/2021, 11:07:19 PM [InsteonLocal] Got updated status for Side Entrance Secondary
5/3/2021, 11:07:19 PM [InsteonLocal] Got updated status for Rear Landscape
5/3/2021, 11:07:20 PM [InsteonLocal] Got updated status for Kitchen Secondary
5/3/2021, 11:07:20 PM [InsteonLocal] Got updated status for Fan
5/3/2021, 11:07:20 PM [InsteonLocal] Getting state for Fan
5/3/2021, 11:07:24 PM [InsteonLocal] Got updated status for Fan
5/3/2021, 11:07:32 PM [InsteonLocal] Got updated status for Hallway Secondary
5/3/2021, 11:07:33 PM [InsteonLocal] Got updated status for Basement Back
5/3/2021, 11:07:35 PM [InsteonLocal] Got updated status for Entrance Secondary
5/3/2021, 11:07:39 PM [InsteonLocal] Got updated status for Basement Front
5/3/2021, 11:07:46 PM [InsteonLocal] Got updated status for Entrance
5/3/2021, 11:07:48 PM [InsteonLocal] Got updated status for Front Landscape


What stands out if that in addition to polling for a 'status', the fan device also polls for a 'state'.

I then changed the device type to 'switch' and restarted the services.  The log now no longer refects polling for both 'status' and 'state', but only 'status'.

5/3/2021, 11:27:41 PM [InsteonLocal] Getting status for Fan

The Home app UI now only reflects ON or OFF and not a %.  Odd thing is, I ran the same test in the past with no change in results.  Not sure if something was cached, or I made an error and didn't actually restart services or something.

Bottom line, it looks like redefining the device as a 'switch' resolved the issue.

**Comment by efthymakis** [comment]
I appreciate your help with this.
